malasakit-django/pcari/admin.py

```diff
@@ -1,313 +1,177 @@
+    list_display = ('respondent', 'get_question_prompt', 'timestamp', 'score')
+
+    # By default first column listed in list_display is clickable; this makes `message` column clickable
+    list_display_links = ('get_question_prompt',)
+
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('timestamp', 'score')
+
+    # Sets fields as readonly
+    readonly_fields = ('respondent', 'question', 'timestamp', 'score')
+
+    # Enables search
+    search_fields = ('question__prompt', 'score')
+
+
+class QuestionAdmin(admin.ModelAdmin):
+    """
+    Abstract admin class for QualitativeQuestionAdmin and QuantitativeQuestionAdmin
+    """
+    # Performance optimizer to limit database queries
+    list_select_related = True
+
+    # Adds a "Save as New" button
+    save_as = True
+        
+
+@admin.register(QualitativeQuestion)
+class QualitativeQuestionAdmin(QuestionAdmin):
+    """
+    Customizes admin change page functionality for QualitativeQuestionAdmin
+    """
+    def get_comments(self, obj):
+        return list(obj.comments)
+
+    # Columns to display in the Comment change list page, in order from left to right 
+    list_display = ('prompt', 'tag', 'get_comments')
+
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('tag', 'prompt')
+
+    # Enables search
+    search_fields = ('prompt', 'tag')
+
+
+@admin.register(QuantitativeQuestion)
+class QuantitativeQuestionAdmin(QuestionAdmin):
+    """
+    Customizes admin change page functionality for QuantitativeQuestionAdmin
+    """
+    # Columns to display in the Comment change list page, in order from left to right
+    list_display = ('prompt', 'tag')
+
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('tag',)
+
+    # Sets fields as readonly
+    readonly_fields = ('prompt', 'tag')
+
+    # Enables search
+    search_fields = ('prompt', 'tag')
+
+
+@admin.register(Respondent)
+class RespondentAdmin(admin.ModelAdmin):
+    """
+    Customizes admin change page functionality for RespondentAdmin
+    """
+    def get_comments(self, obj):
+        return list(obj.comments_made)
+
+    # Empty responses (recorded as None) will be replaced by this placeholder
+    empty_value_display = '-- empty response --'
+
+    # Performance optimizer to limit database queries
+    list_select_related = True
+
+    # Adds a "Save as New" button
+    save_as = True
+
+    # Columns to display in the Comment change list page, in order from left to right
+    list_display = ('get_comments', 'age', 'gender', 'location', 'language', 'submitted_personal_data', 'completed_survey', 'num_questions_rated', 'num_comments_rated')
 
-# class UserProgressionAdmin(admin.ModelAdmin):
-# 	a = 10
-# 	list_display = [a]
+    # Specify which columns we want filtering capabilities for
+    list_filter = ('gender', 'location', 'language', 'submitted_personal_data', 'completed_survey', 'age')
```

Comment:
Remove `age` and `location`--both have too many options to be useful. (The former is the range from 0 to 88, and the latter is free-form, so there is a high amount of variation.)